frontend/src/hooks/useSaveGame.ts

```diff
@@ -0,0 +1,111 @@
+'use client';
+
+import { EventType, trackEvent } from '@/analytics/events';
+import { useApi } from '@/api/Api';
+import { EditGameResponse, isGame } from '@/api/gameApi';
+import { Request, useRequest } from '@/api/Request';
+import useGame from '@/context/useGame';
+import { Game } from '@/database/game';
+import {
+    CreateGameRequest,
+    UpdateGameRequest,
+} from '@jackstenglein/chess-dojo-common/src/database/game';
+import { useSessionStorage } from 'usehooks-ts';
+
+const STAGED_CREATE_GAME_KEY = 'useSaveGame:stageCreateGame';
+
+export interface UseSaveGameFields {
+    createGame: (req: CreateGameRequest) => Promise<void>;
+    updateGame: (req: UpdateGameRequest) => Promise<void>;
+    stageCreateGame: (req: CreateGameRequest) => void;
+    stagedCreateGame: CreateGameRequest | null;
+    request: Request<string>;
+}
+
+export default function useSaveGame(): UseSaveGameFields {
+    const api = useApi();
+    const request = useRequest<string>();
+    const [stagedCreateGame, stageCreateGame] =
+        useSessionStorage<CreateGameRequest | null>(STAGED_CREATE_GAME_KEY, null);
+    const { game } = useGame();
+
+    const createGame = async (createReq: CreateGameRequest) => {
+        request.onStart();
+        let data: Game | EditGameResponse | undefined;
+        try {
+            const response = await api.createGame(createReq);
+            data = response.data;
+        } catch (err) {
+            console.error('CreateGame ', err);
+            request.onFailure(err);
+        }
+
+        if (!data) {
+            return;
+        }
+
+        onCreateGame(createReq, data);
```

Comment:
Why is this section outside the try block?

Reply:
I'd still prefer to have this in the try block. It makes the control flow needlessly complicated to have it outside